### PR TITLE
Fix don't emit shared route error when verb don't match

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix [#2925](https://github.com/microsoft/typespec/issues/2925)

Stop emitting the error if there is a shared route and a non shared route on a different verb.

Also improve the error: 
- change message to be a little more clear
- emit the error on every offending operation not just the first one we find an duplicate

### CODE REVIEW

File: packages/http/src/lib.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ export const $lib = createTypeSpecLibrary({
     "shared-inconsistency": {
       severity: "error",
       messages: {
-        default: "All shared routes must agree on the value of the shared parameter.",
+        default: paramMessage`Route "${"verb"} ${"path"}" is used by 2 operations with different @sharedRoute values.`,
```
Comment (bterlson) [inline packages/http/src/lib.ts:105]:
```suggestion
        default: paramMessage`Route "${"verb"} ${"path"}" is used by multiple operations with different @sharedRoute values.`,
```

It is conceivable the spec has multiple operations, right?

File: packages/http/test/http-decorators.test.ts [hunk 2]
```diff
@@ -271,6 +271,14 @@ describe("http: decorators", () => {
           code: "@typespec/http/shared-inconsistency",
           message: `All shared routes must agree on the value of the shared parameter.`,
         },
+        {
+          code: "@typespec/http/shared-inconsistency",
+          message: `All shared routes must agree on the value of the shared parameter.`,
+        },
+        {
+          code: "@typespec/http/shared-inconsistency",
+          message: `All shared routes must agree on the value of the shared parameter.`,
```
Comment (bterlson) [inline packages/http/test/http-decorators.test.ts:280]: Can we be more specific about what the shared parameter is?

Comment (timotheeguerin) [reply packages/http/test/http-decorators.test.ts:280]: sorry I didn't update this message with the new one, test are failing

File: packages/http/src/lib.ts [hunk 3]
```diff
@@ -102,7 +102,7 @@ export const $lib = createTypeSpecLibrary({
     "shared-inconsistency": {
       severity: "error",
       messages: {
-        default: "All shared routes must agree on the value of the shared parameter.",
+        default: paramMessage`Route "${"verb"} ${"path"}" is used by multiple operations with different @sharedRoute values.`,
```
Comment (bterlson) [inline packages/http/src/lib.ts:105]: I think "@sharedRoute values" is still somewhat confusing. If I understand correctly, maybe something like:

```suggestion
        default: paramMessage`Each operation routed at "${"verb"} ${"path"}" needs to have the @sharedRoute decorator.`,
```

**Review by bterlson — APPROVED** [review]

**Pull request merged**